### PR TITLE
Bump Cython Old

### PR DESCRIPTION
**Pull request by nauaneed**
Old fix from https://github.com/rainwoodman/pandas/blob/05d3fe2402e4563124e7060837ded7513ab5bca7/pandas/_libs/reduction.pyx#L27

### CODE REVIEW

**Comment by nauaneed** [comment]
This is not bad, either.
image
benchmark_20240814_093504

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 requires = [
     "wheel>=0.29.0",
     "setuptools>=42.0.0",
-    "oldest-supported-numpy",
-    "Cython<3.0",
+    "numpy",
```
Comment (prabhuramachandran) [inline pyproject.toml:5]: Is it recommended to use "numpy" instead of "oldest-supported-numpy" now?

**Review by prabhuramachandran — APPROVED** [review]
Looks good to me apart from the change to the `pyproject.toml` which I am not sure about.

File: pyproject.toml (hunk 1 above)
Comment (nauaneed) [reply pyproject.toml:5]: I am not sure if that is recommended, but the problem is this:

requirements.txt and setup.py have just "numpy".  So, `pip install -r requirements.txt` installs the latest numpy. 

Then with `pip install -e .`  cyarray is compiled with oldest-supported-numpy as specified in pyproject.toml. 

Eventually it ends up with, 
```
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
``` 
like here: https://github.com/pypr/cyarray/actions/runs/10434451995/job/28897324886

Comment (prabhuramachandran) [reply pyproject.toml:5]: Hmm, so what is the oldest-supported-numpy version here? Is it < 2.0?

File: cyarray/carray.pyx.mako [hunk 2]
```diff
@@ -37,13 +38,11 @@ The numpy array may however be copied and used in any manner.
...
-IF UNAME_SYSNAME == "Windows":
-    cdef extern from "msstdint.h" nogil:
-        ctypedef unsigned int uintptr_t
-ELSE:
-    from libc.stdint cimport uintptr_t
+
+from libc.stdint cimport uintptr_t
 
 cimport numpy as np
+np.import_array();
```
Comment (prabhuramachandran) [inline cyarray/carray.pyx.mako:45]: Please remove the semicolon.

**Review by prabhuramachandran — COMMENTED** [review]
Just one minor nit, apart from that I am happy to merge this. Thanks!

File: pyproject.toml (hunk 1 above)
Comment (nauaneed) [reply pyproject.toml:5]: Yes, its 1.23.2.

File: cyarray/carray.pyx.mako (hunk 2 above)
Comment (nauaneed) [reply cyarray/carray.pyx.mako]: I have amended this and force pushed now.

Comment (prabhuramachandran) [reply cyarray/carray.pyx.mako]: Thanks!

**Pull request merged**